include/votca/xtp/davidsonsolver.h

```diff
@@ -80,46 +83,43 @@ class DavidsonSolver {
     ProjectedSpace proj = initProjectedSpace(neigen, size_initial_guess);
     RitzEigenPair rep;
 
-    for (Index iiter = 0; iiter < _iter_max; iiter++) {
+    for (_i_iter = 0; _i_iter < _iter_max; _i_iter++) {
+
+      bool do_restart = (proj.search_space() > _max_search_space);
 
-      // restart or update the projection
-      if (proj.search_space > _max_search_space) {
+      if (do_restart) {
         restart(rep, proj, size_initial_guess);
-      } else {
-        updateProjection(A, proj, iiter);
-      }
 
-      // get the ritz vectors
-      switch (this->_matrix_type) {
-        case MATRIX_TYPE::SYMM: {
-          rep = getRitz(proj);
-          break;
-        }
-        case MATRIX_TYPE::HAM: {
-          rep = getHarmonicRitz(A, proj);
+      } else {
+        updateProjection(A, proj);
+
+        // get the ritz vectors
+        switch (this->_matrix_type) {
+          case MATRIX_TYPE::SYMM: {
+            rep = getRitz(proj);
+            break;
+          }
+          case MATRIX_TYPE::HAM: {
+            rep = getHarmonicRitz(A, proj);
+          }
         }
-      }
 
-      // etend the subspace
-      Index nupdate = extendProjection(rep, proj);
+        Index nupdate = extendProjection(rep, proj);
+        proj.V = orthogonalize(proj.V, nupdate);
```

Comment:
ha the orth was even if we do not restart ? better here indeed. Should it be directly in extendProjection ?

Reply:
Yeah maybe, Probably actually